cosmwasm/enclaves/execute/src/registration/attestation.rs

```diff
@@ -289,6 +292,230 @@ pub fn get_mr_enclave() -> [u8; 32] {
+    }
+
+    let qve_isvsvn_threshold: sgx_isv_svn_t = 3;
+    let dcap_ret : sgx_quote3_error_t  = unsafe { sgx_tvl_verify_qve_report_and_identity(
+        vec_quote.as_ptr(),
+        vec_quote.len() as u32,
+        &qe_report,
+        exp_time_s,
+        exp_status,
+        qv_result,
+        p_supp.as_ptr(),
+        n_supp,
+        qve_isvsvn_threshold) };
+
+    if dcap_ret != sgx_quote3_error_t::SGX_QL_SUCCESS {
+        trace!("QVE report verification result: {}", dcap_ret);
+        return Err(sgx_status_t::SGX_ERROR_UNEXPECTED)
+    }
+
+    trace!("n_supp = {}", n_supp);
+    trace!("exp_time_s = {}", exp_time_s);
+    trace!("exp_status = {}", exp_status);
+    trace!("qv_result = {}", qv_result);
+
+    if vec_quote.len() < mem::size_of::<sgx_quote_t>() {
+        trace!("Quote too small");
+        return Err(sgx_status_t::SGX_ERROR_UNEXPECTED);
+    }
+
+    let my_p_quote = vec_quote.as_ptr() as *const sgx_quote_t;
+    let report_body = unsafe { (*my_p_quote).report_body };
+
+    trace!("body.mr_signer = {:?}",  report_body.mr_signer.m);
+    trace!("body.mr_enclave = {:?}", report_body.mr_enclave.m);
+    trace!("body.report_data = {:?}", report_body.report_data.d);
+
+    Ok((report_body, qv_result))
+}
+
+#[cfg(feature = "SGX_MODE_HW")]
+pub fn get_quote_ecdsa(
+    pub_k: &[u8; 32],
+) -> Result<(Vec<u8>, Vec<u8>), sgx_status_t>
+{
+
+    let mut qe_target_info = sgx_target_info_t::default();
+    let mut quote_size: u32 = 0;
+    let mut rt: sgx_status_t = sgx_status_t::default();
+
+    let mut res = unsafe {
+        ocall_get_quote_ecdsa_params(
+            &mut rt as *mut sgx_status_t,
+            &mut qe_target_info,
+            &mut quote_size)
+    };
+
+    if res != sgx_status_t::SGX_SUCCESS {
+        trace!("res = {}", res);
+    }
+
+    if rt != sgx_status_t::SGX_SUCCESS {
+        trace!("rt = {}", rt);
+    }
+
+    trace!("ECDSA quote size = {}", quote_size);
+
+    let mut report_data: sgx_report_data_t = sgx_report_data_t::default();
+    let mut my_report: sgx_report_t = sgx_report_t::default();
+
+    report_data.d[..32].copy_from_slice(pub_k);
+
+    res = unsafe {
+        sgx_create_report(
+            &qe_target_info as *const sgx_target_info_t,
+            &report_data as *const sgx_report_data_t,
+            &mut my_report
+        )
+    };
+
+    if res != sgx_status_t::SGX_SUCCESS {
+        trace!("sgx_create_report = {}", res);
+    }
+
+    let mut vec_quote : Vec<u8> = Vec::new();
+    vec_quote.resize(quote_size as usize, 0);
+
+    res = unsafe {
+
+        ocall_get_quote_ecdsa(
+            &mut rt as *mut sgx_status_t,
+            &my_report,
+            vec_quote.as_mut_ptr(),
+            vec_quote.len() as u32)
+    };
+
+    if res != sgx_status_t::SGX_SUCCESS {
+        trace!("res = {}", res);
+    }
+
+    if rt != sgx_status_t::SGX_SUCCESS {
+        trace!("rt = {}", rt);
+    }
+
+    let mut vec_coll : Vec<u8> = Vec::new();
+    vec_coll.resize(0x4000, 0);
```

Comment:
```rust
let vec_quote: Vec<u8> = vec![0; 0x4000 as usize];
```

Also why 0x4000? Avoid using magic numbers where you can

Reply:
That's a "reasonable" buffer size for the expected result.
If the result won't fit - it'll be truncated, then the code in the enclave would reallocate the needed buffer size and call again.
Using a sane buffer size would spare this redundant call.

Reply:
In this case I would suggest just allocating double the space and calling once - I don't think it's worth the complexity and/or overhead in doing 2 ocalls. The alloc should be fairly cheap comparatively.

Reply:
The size of collateral is not known in advance. Typically it's ~12K, but theoretically can be arbitrary large